src/python/deepgnn/graph_engine/snark/converter/json_converter.py

```diff
@@ -455,27 +455,73 @@ def __add_sparse(node, feature, tp, container):
             + bytes(coordinates.data)
             + values_buf
         )
+
+        if int(k) in container:
+            newId = __get_unassigned_index(node)
+            get_logger().warning(
+                f"Duplicate feature ids found for a node, original: {k}, replaced with: {newId}"
+            )
+            k = newId
+
         container[int(k)] = final_buf
 
 
+def __get_unassigned_index(node):
+    cur_index = node["cur_index"]
+    feature_indexes = node["feature_indexes"]
+    while str(cur_index) in feature_indexes:
+        cur_index += 1
+
+    node["cur_index"] = cur_index + 1
+    node["feature_indexes"].append(cur_index)
+    return cur_index
+
+
 def __add_dense(node, feature, tp, container):
     if feature not in node or node[feature] is None:
         return
     for k in node[feature]:
         values = node[feature][k]
         buf = (tp * len(values))()
         buf[:] = values
-        assert int(k) not in container, "Duplicate feature ids found for a node"
+
+        if int(k) in container:
+            newId = __get_unassigned_index(node)
```

Comment:
nit: please follow [PEP8](https://peps.python.org/pep-0008/#function-and-variable-names), `new_id` instead of `newId`

Reply:
Done, thanks!